src/engine.rs

```diff
@@ -1003,6 +1013,27 @@ impl Reedline {
                 self.enter_history_search();
                 Ok(EventStatus::Handled)
             }
+            ReedlineEvent::DeleteHistoryItem => {
+                if self.input_mode == InputMode::HistoryTraversal {
+                    match self
+                        .history_cursor
+                        .delete_item_at_cursor(self.history.as_mut())
+                    {
+                        Some(Ok(())) => {
+                            self.update_buffer_from_history();
+                            // are these needed/correct?
+                            self.editor.move_to_start(UndoBehavior::HistoryNavigation);
+                            self.editor
+                                .move_to_line_end(UndoBehavior::HistoryNavigation);
```

Comment:
I am not quite sure why they both appear for `previous_history` but only one for `next_history`. I can not immediately recall what problem with undo and redo might have prompted it.

Worth thinking about and playing with. This operation is irreversible on the history. Will the undo/redo move leave the buffer in a broken state?

Reply:
I didn't look closely enough, it's `move_to_line_end` vs `move_to_end` from [this change](https://github.com/nushell/reedline/pull/306/files#diff-ff278ec599adfd033d35a083ebe8948db8bf177c9817dee8ddc4f3a9e3f5728eR831) to deal with multiline entries. I've added a comment.